### PR TITLE
Update cmsplit_type.c to add tests for communicator unguided split case

### PR DESCRIPTION
**Pull request by drwootton**
Update the cmsplit_type.c  testcase to add tests for the unguided communicator split 
(MPI_COMM_TYPE_HW_UNGUIDED) case.

Signed-off-by: David Wootton <dwootton@us.ibm.com>

### CODE REVIEW

**Comment by gpaulsen** [comment]
Requires this PR to pass: https://github.com/open-mpi/ompi/pull/10739

**Comment by jsquyres** [comment]
Rebased to get the new GitHub Action CI test.

File: comm_split_type/cmsplit_type.c [hunk 1]
```diff
@@ -183,6 +191,7 @@ int main(int argc, char *argv[])
 
     sync_hr();
 
+#if 0
```
Comment (jjhursey) [inline comm_split_type/cmsplit_type.c:194]: Why is this `MPI_COMM_TYPE_HW_GUIDED` test commented out? It should work.

**Review by jjhursey — CHANGES_REQUESTED** [review]
1 note but otherwise it looks fine.

**Review by jjhursey — APPROVED** [review]

**Pull request merged**